builtin/credential/aws-ec2/path_config_certificate.go

```diff
@@ -240,19 +342,31 @@ func (b *backend) pathConfigCertificateRead(
 	}, nil
 }
 
-// pathConfigCertificateCreateUpdate is used to register an AWS Public Key that is
-// used to verify the PKCS#7 signature of the instance identity document.
+// pathConfigCertificateCreateUpdate is used to register an AWS Public Key that
+// is used to verify the PKCS#7 signature of the instance identity document.
 func (b *backend) pathConfigCertificateCreateUpdate(
 	req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
 	certName := data.Get("cert_name").(string)
 	if certName == "" {
-		return logical.ErrorResponse("missing cert_name"), nil
+		return logical.ErrorResponse("missing certificate name"), nil
+	}
+
+	certType := data.Get("type").(string)
```

Comment:
I think for backwards compatibility it would be good to default this to pkcs7 (see my earlier comment) but perhaps return a warning instead letting the user know that they should set the type explicitly.